streams/test-utils/src/main/java/org/apache/kafka/streams/test/OutputVerifier.java

```diff
@@ -352,4 +360,79 @@
         Objects.requireNonNull(expectedRecord);
         compareKeyValueHeaders(record, expectedRecord.key(), expectedRecord.value(), expectedRecord.headers());
     }
+
+    /**
+     * Compares a {@link ProducerRecord} with the provided key, value, and headers and throws an
+     * {@link AssertionError} if the {@code ProducerRecord}'s key, value, or headers is not equal to the expected key,
+     * value, or headers.
+     *
+     * @param record a output {@code ProducerRecord} for verification
+     * @param expectedKey the expected key of the {@code ProducerRecord}
+     * @param expectedValue the expected value of the {@code ProducerRecord}
+     * @param expectedHeaders the expected headers of the {@code ProducerRecord}
```

Comment:
`expectedTimestamp` parameter missing (insert before `expectedHeaders` to align with method name.